### ISSUE TITLE
UT projection path does not accumulate 2D mean gradients

### ISSUE THREAD

**Issue by harrism** [openvdb/fvdb-core#609]
### Description

When `accumulate_mean_2d_gradients` is `True`, the ANALYTIC projection path in `GaussianSplat3d._do_projection` lazily initializes and accumulates into `_accumulated_gradient_step_counts` and `_accumulated_mean_2d_gradient_norms` during forward/backward passes. The UNSCENTED (UT) projection path does not — it calls `_C.project_gaussians_ut_fwd` which does not receive or update these tensors.

This means that for any camera model that uses the UT path (all OpenCV variants, or explicit `ProjectionMethod.UNSCENTED`), the gradient accumulation tensors exist but remain zero. Downstream consumers that rely on accumulated gradient data to make refinement decisions (e.g., adaptive densification in fvdb-reality-capture) effectively get no gradient signal — no Gaussians are ever duplicated or split.

### Current state

PR #608 fixed the initialization bug (the tensors were `None` because the lazy init block ran after the UT early return). After that fix, the tensors are properly initialized to zero — but they are never updated by the UT kernel, so they stay zero for the entire training run.

### Impact

Any COLMAP dataset with non-pinhole cameras (SIMPLE_RADIAL, RADIAL, OPENCV — which is the majority of real-world COLMAP reconstructions) uses the UT path and gets no adaptive densification. The workaround is to undistort images first (`colmap image_undistorter`) to convert to PINHOLE cameras, but this should not be required.

### Observed behavior

Reconstruction with the strawberry dataset (OPENCV cameras → UT path):

```
Refinement (step 922):  0 duplicated, 0 split, 4,948 pruned  (324,027 → 319,079)
Refinement (step 1383): 0 duplicated, 0 split, 5,601 pruned  (319,079 → 313,478)
```

Same dataset after `colmap image_undistorter` (PINHOLE cameras → ANALYTIC path):

```
Refinement (step 922):  39,570 duplicated, 33,266 split, 3,272 pruned  (324,027 → 393,591)
Refinement (step 1383): 101,779 duplicated, 50,388 split, 3,396 pruned (393,591 → 542,362)
```

### Expected behavior

The UT projection path should accumulate gradient data the same way the ANALYTIC path does, so adaptive densification works for all camera models.

### Possible approach

The UT forward kernel (`_C.project_gaussians_ut_fwd`) would need to accept and update the accumulation tensors, or the accumulation could be done in the Python wrapper after the C++ call returns (using `means2d` gradients in the backward pass, similar to how `_ProjectGaussiansFn` handles it for the ANALYTIC path).

### Related

- https://github.com/openvdb/fvdb-reality-capture/issues/279 — the original crash report
- https://github.com/openvdb/fvdb-core/pull/608 — fix for the initialization bug (tensors were None)
- https://github.com/openvdb/fvdb-reality-capture/pull/281 — defensive guard in fvdb-reality-capture

**Comment by harrism** [comment]
It was pointed out by @fwilliams that the UT path cannot accumulate gradients because the projection is nondifferentiable. Thus the MCMC optimizer is a requirement for distorted cameras. So I'm closing this issue and will file another one for fvdb-reality-capture about improving the UX around this use case (e.g. erring out with inappropriate combinations).
